lib/wasm-contract/src/contract.rs

```diff
@@ -0,0 +1,135 @@
+            if base.imports.contains_key(&key) {
+                if val != base.imports[&key] {
+                    return Err(format!("Conflict detected: the key {} was found in imports but the definitions were different: {:?} {:?}", key, base.imports[&key], val));
+                }
+            } else {
+                let res = base.imports.insert(key, val);
+                debug_assert!(res.is_none());
+            }
+        }
+
+        for (key, val) in other.exports.into_iter() {
+            if base.exports.contains_key(&key) {
+                if val != base.exports[&key] {
+                    return Err(format!("Conflict detected: the key {} was found in exports but the definitions were different: {:?} {:?}", key, base.exports[&key], val));
+                }
+            } else {
+                let res = base.exports.insert(key, val);
+                debug_assert!(res.is_none());
+            }
+        }
+        Ok(base)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Import {
+    Func {
+        namespace: String,
+        name: String,
+        params: Vec<WasmType>,
+        result: Vec<WasmType>,
+    },
+    Global {
+        name: String,
+        var_type: WasmType,
+    },
+}
+
+// TODO: figure out this separator... '/' is a valid character in names so we can have collisions
+impl Import {
+    /// Get the key used to look this import up in the Contract's import hashmap
+    pub fn get_key(&self) -> String {
+        match self {
+            Import::Func {
+                namespace, name, ..
+            } => format!("{}/{}", &namespace, &name),
+            Import::Global { name, .. } => name.clone(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Export {
+    Func {
+        name: String,
+        params: Vec<WasmType>,
+        result: Vec<WasmType>,
+    },
+    Global {
+        name: String,
+        var_type: WasmType,
+    },
+}
+
+impl Export {
+    /// Get the key used to look this export up in the Contract's export hashmap
+    pub fn get_key(&self) -> String {
+        match self {
+            Export::Func { name, .. } => name.clone(),
+            Export::Global { name, .. } => name.clone(),
+        }
+    }
+}
+
+/// Primitive wasm type
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum WasmType {
+    I32,
+    I64,
+    F32,
+    F64,
+}
+
+#[cfg(test)]
+mod test {
+    use crate::parser;
+
+    #[test]
+    fn merging_works() {
+        let contract1_src = r#"(assert_import (func "env" "plus_one" (param i32) (result i32)))"#;
+        let contract2_src = r#"(assert_import (func "env" "plus_one" (param i64) (result i64)))"#;
+        let contract3_src = r#"(assert_import (func "env" "times_two" (param i64) (result i64)))"#;
+        let contract4_src =
+            r#"(assert_import (func "env" "times_two" (param i64 i64) (result i64)))"#;
+        let contract5_src = r#"(assert_export (func "empty_bank_account" (param) (result)))"#;
```

Comment:
When having a func that takes no params, the WebAssembly text spec just skips it:

```wasm
(module
  (func $getAnswer (result i32)
    i32.const 42)
)
```

And same for result
```wasm
(module
  (func $log (param i32))
)
```


I think we can do the same. Thoughts?